src/cozmo_companion/assistant.py

```diff
@@ -103,47 +127,82 @@ def _listen(self):
+        returns an appropriate inquiry string based on the content.
+
+        Args:
+            user_text (str): The text provided by the user.
+
+        Returns:
+            Optional[str]: The feedback inquiry string or None if no match is found.
+        """
+        # List of feedback phrases to check against
+        feedback_phrases = ["joke", "motivational quote"]
+        for phrase in feedback_phrases:
+            # If the user's text contains any of the feedback phrases, return the inquiry string
+            if phrase in user_text.lower():
+                return " Did this help put a smile to your face?"
+        # If no match is found, return None
+        return None
+
+    def construct_gpt_prompt(self, text):
+        """Construct the GPT-3 prompt based on the user's sentiment."""
+        # Detect the sentiment of the user's text
+        detected_sentiment = Sentiment(text)
+        # If the sentiment is negative, append a request for an empathetic response
+        if detected_sentiment == Sentiment.NEGATIVE:
+            print("Negative Sentiment Detected...")
+            return text + " Requesting an empathetic response."
+        return text
+
     def _generate_response(self, text):
         """Generate a GPT response to the user's text input."""
-
         try:
-            # Add the user's input to the Chat GPT history log
-            self.conversation_history.append({"role": "user", "content": text})
+            # Construct the GPT prompt based on the input text
+            gpt_prompt = self.construct_gpt_prompt(text)
 
-            gpt_response = self.chatbot(text)
+            # Update the conversation history with the user's input
+            self.conversation_history.append({"role": "user", "content": gpt_prompt})
 
-            gpt_response_msg = gpt_response.content
+            # Get the chatbot's response content
+            gpt_response = self.chatbot(gpt_prompt).content
 
-            self.conversation_history.append(
-                {
-                    "role": "gpt",
-                    "content": gpt_response_msg,
-                }
-            )
+            # If there's a feedback inquiry, append it to the chatbot's response
+            feedback_inquiry = self.get_feedback_inquiry(gpt_prompt)
+            if feedback_inquiry and feedback_inquiry not in gpt_response:
+                gpt_response += feedback_inquiry
 
+            # Update the conversation history with the chatbot's response
+            self.conversation_history.append({"role": "gpt", "content": gpt_response})
+
+            # Print the conversation log (this can be removed if not needed in production)
             print("Conversation Log: ", self.conversation_history)
 
-            return gpt_response_msg
+            return gpt_response
         except Exception as e:
             logging.error(f"Error getting GPT completion: {e}", exc_info=True)
             return "I'm sorry, I couldn't process that."
```

Comment:
excepting so generally is normally discouraged.

However, in this case it seems OK, as you log the error

Reply:
I guess this is why we see these eventloop errors in the output? 

I suggest to create a follow-up issue to investigate that error (low prio)